rerun_cpp/src/rerun/recording_stream.hpp

```diff
@@ -868,34 +870,87 @@ namespace rerun {
         ///
         /// Any failures that may occur during serialization are handled with `Error::handle`.
         ///
+        /// \param entity_path The path to the recording property.
+        /// \param properties The properties of the recording.
+        /// \see `try_set_properties`
+        template <typename... Ts>
+        void set_properties_with_prefix(std::string_view entity_path, const Ts&... properties)
+            const {
+            try_set_properties_with_prefix(entity_path, properties...).handle();
+        }
+
+        /// Set the properties of a recording.
+        ///
+        /// \param entity_path The path to the recording property.
+        /// \param properties The properties of the recording.
+        /// \see `set_properties`
+        template <typename... Ts>
+        Error try_set_properties_with_prefix(std::string_view entity_path, const Ts&... properties)
+            const {
+            rr_error status = {};
+            this->log_static(
+                this->PARTITION_PROPERTIES_ENTITY_PATH + "/" + std::string(entity_path),
+                properties...
+            );
+            return status;
+        }
+
+        /// Set the properties of a recording.
+        ///
+        /// This is a shorthand method for logging to the root of the recording properties.
+        ///
         /// \param properties The properties of the recording.
-        /// \see `try_set_recording_properties`
-        void set_properties(archetypes::RecordingProperties& properties) const {
-            try_set_properties(properties).handle();
+        /// \see `try_set_properties`
+        template <typename... Ts>
+        void set_properties(const Ts&... properties) const {
+            // We need to split this into two arguments so that the compiler can bottom out.
+            try_set_properties_with_prefix("/", properties...).handle();
```

Comment:
consequence of above is 
```suggestion
            try_set_properties_with_prefix("", properties...).handle();
```